scripts/cleanOutdatedFrameworkProductions.py

```diff
@@ -0,0 +1,208 @@
+                t = str(protectedtag).split('_%s_' % part)
+                if t[0] in tags and t[1] in tags:
+                    if DEBUG:
+                        print '\tSkipping whitelisted sample %s' % s_name
+                    isSampleProtected = True
+            if not isSampleProtected:
+                tags = tags.replace(part, '') # remove HHAnalyzer and the like from the name of the sample
+        if isSampleProtected:
+            continue
+        # now extract the fw and analyzer tags
+        # for analyzer, this is always the last part of the sample name so we don't have to worry about naming conventions there (fortunately)
+        tags = tags.split('_')
+        Anatag = tags[-1]
+        tags = tags[:-1]
+        # for FW the following regex should work ((v\d+.\d+.\d+\+\d+X?)(-\d+-g[0-9a-f]{7,40})?)|([0-9a-f]{7,40})
+        # it matches either:
+        #   - a framework tag (possibly with a final X): v1.2.0+7415
+        #   - possibly followed by a number of commits and a 'g' plus 7 to 40 characters git hash: v1.2.0+7415-79-ga5b16ff
+        #   - or alternatively a 7 to 40 characters git hash: f2f0a44
+        tags = [x for x in tags if re.match('((v\d+.\d+.\d+\+\d+X?)(-\d+-g[0-9a-f]{7,40})?)|([0-9a-f]{7,40})', x)]
+        if DEBUG:
+            print tags, Anatag
+        if len(tags) != 1:
+            print "ERROR, there are spurious things in the sample name, please figure out what is happening:"
+            print "FWtags= ", tags
+            return 1
+        FWtag = tags[0]
+        list_all_productions.append([FWtag, Anatag])
+
+    for i, p in enumerate(list_all_productions):
+        if DEBUG and i > 0:
+            break
+        FWtag, Anatag = p
+
+        extrastring = ''
+        if not evaluateSize:
+            extrastring = '(evaluation of the disk size is OFF by default)'
+        print "\n##### Now looking at prod FWtag= ", FWtag, 'Anatag= ', Anatag, 'and list the associated folders %s' % extrastring
+        totalSize = 0
+        totalSamples = 0
+        for s_name, s_id in list_allDBsamples:
+            if FWtag in str(s_name) and Anatag in str(s_name):
+                result = dbstore.find(Sample, Sample.name == s_name)
+                s = result.one()
+                if evaluateSize:
+                    totalSize += int(subprocess.check_output(["du", '-s', str(s.path)]).split()[0].decode('utf-8'))
+                totalSamples += 1
+                if s.source_sample is not None:
+                    print "WARNING, the sample", s_name, "depend on another sample, aborting now"
+                    return 2
+                if s.derived_samples.count() > 0:
+                    print "WARNING, the sample", s_name, "has derived samples, aborting now"
+                    return 2
+                if s.results.count() > 0:
+                    print "WARNING, the sample", s_name, "has derived results, aborting now"
+                    return 2
```

Comment:
Maybe it's better to skip the current production and goes to the next one instead of exiting everything? (same for the 2 returns above)